### PR TITLE
Allow capacitors to recharge Baubles items

### PR DESCRIPTION
**Pull request by KaseiFR**

### CODE REVIEW

File: src/main/java/cofh/thermalexpansion/item/ItemCapacitor.java [hunk 1]
```diff
@@ -167,6 +177,22 @@ public void onUpdate(ItemStack stack, World world, Entity entity, int slot, bool
 		}
 	}
 
+	private static Iterable<ItemStack> getBaubles(Entity entity) {
+		if (CAPABILITY_BAUBLES == null) {
```
Comment (covers1624) [inline src/main/java/cofh/thermalexpansion/item/ItemCapacitor.java]: New line after method deceleration.

File: src/main/java/cofh/thermalexpansion/item/ItemCapacitor.java [hunk 2]
```diff
@@ -167,6 +177,22 @@ public void onUpdate(ItemStack stack, World world, Entity entity, int slot, bool
...
+
+		IBaublesItemHandler handler = entity.getCapability(CAPABILITY_BAUBLES, null);
+		if (handler == null) {
+			return Collections.emptyList();
+		}
+
+		return IntStream.range(0, handler.getSlots())
+				.mapToObj(handler::getStackInSlot)
+				.filter(stack -> !stack.isEmpty())
+				.collect(Collectors.toList());
```
Comment (covers1624) [inline src/main/java/cofh/thermalexpansion/item/ItemCapacitor.java]: probably add some ninja comments here so our formatters don't one line it again.

**Review by covers1624 — CHANGES_REQUESTED** [review]
Looks good to me, some minor style changes are needed.

**Comment by KaseiFR** [comment]
Done. I don't know the param of your formatter, so I hope it will like it this way.

**Comment by covers1624** [comment]
You'll have to whack one on each line, just a simple `//` should do, unless lemming is fine with it being smushed to one line and we drop the ninja comments all together.

**Comment by KaseiFR** [comment]
I've added the fake comments

**Comment by KingLemming** [comment]
This is only a soft dep right? TE won't depend on Baubles being present?

**Review by covers1624 — APPROVED** [review]

**Comment by miuirussia** [comment]
Looks like it has hard dependency for Baubles

**Comment by KaseiFR** [comment]
No, it's not a hard dependency, and I've tested it.  
If Baubles is not present, the capability is never injected and the code depending on Baubles is never reached.

**Comment by covers1624** [comment]
Hmm, i was sure that the JVM would try and define every class used in a method when called. Lemme test this myself.

**Comment by covers1624** [comment]
Well seems i was wrong. Good to merge.

**Comment by KaseiFR** [comment]
Can't blame you, the JVM is weird. And it's the first time I found a positive side to type erasure: I think I can only manipulate the capability because its generic type has been erased.

**Comment by covers1624** [comment]
Yeah, The generic isn't erased, its essentially just metadata for compile time and reflection, the JVM doesn't care about it when defining a class.

**Comment by KingLemming** [comment]
Alright. We're gonna try this out.

**Comment by KingLemming** [comment]
@KaseiFR - do you have a twitter handle?

**Comment by KaseiFR** [comment]
Nope

**Pull request merged**